### PR TITLE
feat: vllm integration

### PR DESCRIPTION
**Pull request by jorgeantonio21**

### CODE REVIEW

File: backends/vllm/src/block.rs [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+    /// * `Ok(())` if the tokens were successfully appended.
+    /// * `Err(BlockError::AllocationError)` if there isn't enough space in the block.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// let mut block = LogicalTokenBlock::new(0, 10);
+    /// assert!(block.append_tokens(&[1, 2, 3]).is_ok());
+    /// assert!(block.append_tokens(&[4, 5, 6, 7, 8, 9, 10]).is_err());
+    /// ```    #[instrument(skip_all)]
```
Comment (Cifko) [inline backends/vllm/src/block.rs:102]: The instrument should not be in the comment.

File: backends/vllm/src/block.rs [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+    ///
+    /// * `Ok(())` if the tokens were successfully appended.
+    /// * `Err(BlockError::AllocationError)` if there isn't enough space in the block.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// let mut block = LogicalTokenBlock::new(0, 10);
+    /// assert!(block.append_tokens(&[1, 2, 3]).is_ok());
+    /// assert!(block.append_tokens(&[4, 5, 6, 7, 8, 9, 10]).is_err());
```
Comment (Cifko) [inline backends/vllm/src/block.rs:101]: Appending 7 new tokens to 3 old ones, for blocks_size 10, should still be fine.

**Review by Cifko — APPROVED** [review]

File: backends/vllm/src/block.rs (hunk 1 above)
Comment (jorgeantonio21) [reply backends/vllm/src/block.rs:102]: Great catch up !

File: backends/vllm/src/block.rs (hunk 2 above)
Comment (jorgeantonio21) [reply backends/vllm/src/block.rs]: Great catch up!

**Pull request merged**